.github/workflows/publish.yml

```diff
@@ -0,0 +1,53 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the repository
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      # Set up JDK
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+
+      # Set Gradle Properties as Environment Variables
+      # You can use any gradle property in the gradle.properties file as an environment variable.
+      # eg: ballerinaVersion=2.0.0 -> gradle.ballerinaVersion
+      - name: Set Gradle Properties as Environment Variables
+        run: |
+          while IFS='=' read -r key value; do
+            [[ -z "$key" || "$key" == \#* ]] && continue
+            key=$(echo $key | xargs)
+            value=$(echo $value | xargs)
+            prefixed_key="gradle_$key"
+            echo "$prefixed_key=$value" >> $GITHUB_ENV
+            echo "Exported $prefixed_key as an environment variable."
+            echo "Value: $value"
+          done < gradle.properties
+
+      # Install Ballerina
+      - name: Install Ballerina
+        uses: ballerina-platform/setup-ballerina@v1
+        with:
+          version: ${{ env.gradle_ballerinaLangVersion }}
```

Comment:
Better to set env variable names with uppercase letters

https://github.com/ballerina-platform/ballerina-library/blob/89d4003f0083623c88cf7ec4b4ad96c0fe2a2326/.github/workflows/pr-build-connector-template.yml#L39

Reply:
Made them uppercase